pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -101,6 +106,68 @@ type CiliumBGPNeighborGracefulRestart struct {
+type BGPStandardCommunity string
+
+// BGPLargeCommunity type represents a value of the BGP Large Communities Attribute (RFC 8092),
+// as three 4-byte decimal numbers separated by colons.
+// +kubebuilder:validation:Pattern=`^([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5])$`
+type BGPLargeCommunity string
+
+// BGPCommunities holds community values of the supported BGP community path attributes.
+type BGPCommunities struct {
+	// Standard holds a list of "standard" 32-bit BGP Communities Attribute (RFC 1997) values.
+	//
+	// +kubebuilder:validation:Optional
+	Standard []BGPStandardCommunity `json:"standard,omitempty"`
+
+	// Large holds a list of the BGP Large Communities Attribute (RFC 8092) values.
+	//
+	// +kubebuilder:validation:Optional
+	Large []BGPLargeCommunity `json:"large,omitempty"`
+}
+
+// CiliumBGPPathAttributes can be used to apply additional path attributes
+// to matched routes when advertising them to a BGP peer.
+type CiliumBGPPathAttributes struct {
+	// SelectorType defines the object type on which the Selector applies:
+	// - For "PodCIDR" the Selector matches k8s CiliumNode resources
+	//   (path attributes apply to routes announced for PodCIDRs of selected CiliumNodes.
+	//   Only affects routes of cluster scope / Kubernetes IPAM CIDRs, not Multi-Pool IPAM CIDRs.
+	// - For "CiliumLoadBalancerIPPool" the Selector matches CiliumLoadBalancerIPPool custom resources
+	//   (path attributes apply to routes announced for selected CiliumLoadBalancerIPPools).
+	//
+	// +kubebuilder:validation:Enum=PodCIDR;CiliumLoadBalancerIPPool
+	// +kubebuilder:validation:Required
+	SelectorType string `json:"selectorType"`
+
+	// Selector selects a group of objects of the SelectorType
+	// resulting into routes that will be announced with the configured Attributes.
+	// If nil / not set, all objects of the SelectorType are selected.
+	//
+	// +kubebuilder:validation:Optional
+	Selector *slimv1.LabelSelector `json:"selector,omitempty"`
+
+	// Communities defines a set of community values advertised in the supported BGP Communities path attributes.
+	// If nil / not set, no BGP Communities path attribute will be advertised.
+	//
+	// +kubebuilder:validation:Optional
+	Communities *BGPCommunities `json:"communities,omitempty"`
```

Comment:
What is the expected behavior if `Communities` is defined but `Standard` and `Large` is `nil` since they're both optional? If this is unexpected, add some validation logic to the reconcile loop. I think this can be done in a follow-on PR if you agree.